core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala

```diff
@@ -221,6 +221,32 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
       }
     }
 
+    val globalTempViews: mutable.Buffer[SnappyGlobalTemporaryViewStats] = {
+
+      val snc = SnappyContext.apply(this.sparkContext)
+      val snappySessionCatalog = snc.snappySession.sessionCatalog
+      val globalTempSchema: String = "global_temp"
+      val gblTempViewList = snappySessionCatalog.listTables(globalTempSchema)
+
+      if (gblTempViewList.isEmpty) {
```

Comment:
Temporarily creating a session just for listing the tables is quite inefficient. Can't this be included as part of regular tableTypesMap created in getStatsFromAllServers (where tableType can be GLOBAL_TEMPORARY)?

To get the list without session use: SnappyContext.sharedState(sc = null).globalTempViewManager.listViewNames("*"). This avoids unnecessary creation of a SnappyContext object which is expensive and not required. Secondly this list can just be appended to the list of "hiveTables" in getStatsFromAllServers with tableType=GLOBAL_TEMPORARY. Display can use the tableType to distinguish. This avoids separate calls for this as well as the new message.